### ISSUE TITLE
systemd-networkd: DHCPv4 won't acquire IP from Google Fiber on 255-rc1

### ISSUE THREAD

**Issue by mamarley** [systemd/systemd#29904]
### systemd version the issue has been seen with

255-rc1

### Used distribution

Kubuntu 24.04

### Linux kernel version used

6.6.0

### CPU architectures issue was seen on

x86_64

### Component

systemd-networkd

### Expected behaviour you didn't see

systemd-networkd should acquire an IPv4 address as it did with previous versions.

### Unexpected behaviour you saw

systemd-networkd fails to acquire an IPv4 address.  The following log output is produced when debug logging is turned on:
```
systemd-networkd[228649]: wan: Requested configuring of the DHCPv4 client.
systemd-networkd[228649]: wan: DHCPv4 client: STARTED on ifindex 2
systemd-networkd[228649]: wan: DHCPv4 client is configured, acquiring DHCPv4 lease.
systemd-networkd[228649]: wan: DHCPv4 client: DISCOVER
systemd-networkd[228649]: wan: DHCPv4 client: State changed: initialization -> selecting
systemd-networkd[228649]: wan: DHCPv4 client: Received message from RAW socket, processing.
systemd-networkd[228649]: wan: DHCPv4 client: received OFFER with Rapid Commit option, ignoring
```
followed by an endless loop of:
```
systemd-networkd[228649]: wan: DHCPv4 client: DISCOVER
systemd-networkd[228649]: wan: DHCPv4 client: Received message from RAW socket, processing.
systemd-networkd[228649]: wan: DHCPv4 client: received OFFER with Rapid Commit option, ignoring
```
It would appear as if Google Fiber's DHCPv4 server is behaving in a standards-incompliant manner and sending an OFFER message with the Rapid Commit option, which breaks with the new Rapid Commit feature in 255.  This occurs regardless of whether RapidCommit is turned on or off.  I tried commenting out https://github.com/systemd/systemd/blob/main/src/libsystemd-network/sd-dhcp-client.c#L1612 through line 1614 and that resolves the issue.

The configuration of the relevant interface is:
```
[Match]
Name=wan

[Network]
DHCP=yes

[DHCPv4]
UseDNS=no
UseHostname=no
UseNTP=no
SendRelease=no

[DHCPv6]
WithoutRA=solicit
PrefixDelegationHint=::/56
UseDNS=no
UseNTP=no
SendRelease=no

[IPv6AcceptRA]
UseDNS=no
UseDomains=no

[Link]
RequiredForOnline=routable
RequiredFamilyForOnline=both
```

### Steps to reproduce the problem

Use systemd-networkd 255 to attempt to get a DHCPv4 lease from Google Fiber.

**Comment by bluca** [comment]
Please add full debug level log of networkd

**Comment by yuwata** [comment]
Oh, thank you for testing RC1. I will submit a patch.

**Comment by mamarley** [comment]
Here's the full log: https://pastebin.ubuntu.com/p/Hs3QsR3mNr/

**Comment by yuwata** [comment]
Fix is waiting in #29908. @mamarley Could you test the patch?
